attendance/API_HR/c-hr.php

```diff
@@ -2302,6 +2302,138 @@ public static function updateGooglepaySlipDriveAccessToken($google_access_token)
+                $mail->addReplyTo('hr@excellencetechnologies.in', 'Excellence Technologies'); // reply email address with name 
+                $mail->addAddress($work_email, $name); // name and address to whome mail is to send
+                if (sizeof($cc) > 0) {
+                    foreach ($cc as $d) {
+                        $mail->addCC($d[0], $d[1]);
+                    }
+                }
+                if (sizeof($bcc) > 0) {
+                    foreach ($bcc as $d2) {
+                        $mail->addBCC($d2[0], $d2[1]);
+                    }
+                }
+                $mail->Subject = $subject; // subject of email message 
+                $mail->msgHTML($body); // main message 
+                // $mail->AltBody = 'This is a plain-text message body';
+                //Attach an image file
+                if (sizeof($file_upload) > 0) {
+                    foreach ($file_upload as $d3) {
+                        $mail->addAttachment($d3);
+                    }
+                }
+//send the message, check for errors
+                if (!$mail->send()) {
+                    $row3 = $mail->ErrorInfo;
+                } else {
+                    $row3 = "Message sent";
+                }
+            }
+        }
+
+        
+        if ($row3 != "Message sent") {
+
+            $r_error = 1;
+            $r_message = $row3;
+            $r_data['message'] = $r_message;
+        } else {
+            $r_error = 0;
+            $r_message = "Message Sent";
+            $r_data['message'] = $r_message;
+        }
+        $return = array();
+        $return['error'] = $r_error;
+        $return['data'] = $r_data;
+        return $return;
+    }
+
+
+
+
+    //New Employee Welcome Email
+
+    public function sendWelcomeMail($userID) {
+        $userInfo = self::getUserInfo($userID);
+
+        // Fetching welcome email template
+        $q = "SELECT * FROM email_templates where name='Welcome mail'";
+        $runQuery = self::DBrunQuery($q);
+        $row = self::DBfetchRows($runQuery);
+        if(empty($row)) {
+           echo "Template not fetched!";
+           die; 
+        }
+        $mail_body = $row[0]['body'];
+        
+        $work_email = $userInfo['work_email'];
+        $replace_to = array();
+        $replace_to[0] = $userInfo['name'];
+        $replace_to[1] = $userInfo['dateofjoining'];
+        $replace_from = array('#employee_name','#joining_date');
+
+        $mail_body = str_replace($replace_from,$replace_to,$mail_body);
+
+        // Fetching value of Variables in Template
+        $q2 = 'Select * from template_variables';
+        $runQuery2 = self::DBrunQuery($q2);
+        $row2 = self::DBfetchRows($runQuery2);
+        foreach ($row2 as $s) {
+            $mail_body = str_replace($s['name'], $s['value'], $mail_body);
+        }
+
+        $data = array();
+        $data['email']['subject'] = 'Welcome to Excellence Technologies';
```

Comment:
can u see and check , i think subject also comes from template? if yes the no need to hard code here.